x-pack/plugins/endpoint/public/applications/endpoint/store/index.ts

```diff
@@ -66,7 +47,7 @@ export const appStoreFactory: (middlewareDeps?: {
    * Any additional Redux Middlewares
    * (should only be used for testing - example: to inject the action spy middleware)
    */
-  additionalMiddleware?: Array<ReturnType<MiddlewareFactory>>;
+  additionalMiddleware?: Array<ReturnType<ImmutableMiddlewareFactory>>;
```

Comment:
@paul-tavares objections to replacing this with:
```
actionSpyMiddleware?: ActionSpyMiddleware
```
I don't think code outside of the store should dictating what middleware are used, how they are constructed, or what order they are applied in.